src/turtle_setter.cpp

```diff
@@ -0,0 +1,186 @@
+{
+    const float DEGREES_IN_QUADRANT = 90;
+    float arctangent = atan(x/y)*M_PI;
+    if(y > 0 && x < 0)
+    {
+        arctangent = -arctangent+DEGREES_IN_QUADRANT*1; //Quadrant 2
+    }
+    else if(y < 0 && x < 0)
+    {
+       arctangent = arctangent+DEGREES_IN_QUADRANT*2; //Quadrant 3
+    }
+    else if(y < 0 && x > 0)
+    {
+        arctangent = -arctangent+DEGREES_IN_QUADRANT*3; //Quadrant 4
+    }
+
+    return arctangent;
+}
+
+void calculateDistance(Waypoint targetWaypoint, Waypoint currentWaypoint, Waypoint &resultWaypoint)
+{
+    resultWaypoint.x = targetWaypoint.x - currentWaypoint.x;
+    resultWaypoint.y = targetWaypoint.y - currentWaypoint.y;
+    resultWaypoint.distance = distance(resultWaypoint.x, resultWaypoint.y);
+    resultWaypoint.angle = arctan(resultWaypoint.x, resultWaypoint.y);
+}
+
+void updateDistanceMsg(ros::NodeHandle &n, Waypoint DistanceStruct)
+{
+    ros::Publisher publisher = n.advertise<software_training_assignment::distance>("Distances",1000);
+    software_training_assignment::distance msg;
+    msg.x = DistanceStruct.x;
+    msg.y = DistanceStruct.y;
+    msg.dist = DistanceStruct.distance;
+    publisher.publish(msg);
+}
+
+
+/******CLEAR & SPAWN TURTLE FUNCTIONS******/
+bool clearTurtle(std::string name) //clears the existing turtle
+{
+    turtlesim::Kill killService; //Service to kill the turtle
+    killService.request.name = name;
+    return ros::service::call("kill", killService); //Makes sure the turtle dies before exiting the loop
+
+}
+
+
+bool spawnTurtle(std::string name, double x, double y, double angle)
+{
+    turtlesim::Spawn spawnService;
+    spawnService.request.name = name;
+    spawnService.request.x = x;
+    spawnService.request.y = y;
+    spawnService.request.theta = angle;
+    return ros::service::call("spawn", spawnService);
+}
+
+/******CLEAR & SPAWN TURTLE FUNCTIONS END******/
+
+bool resetTurtlePosition(software_training_assignment::resetTurtle::Request &request, software_training_assignment::resetTurtle::Response &response)
+{
+    const double RESET_X = 25;
+    const double RESET_Y = 10;
+    std::string name = request.name;  //QUESTION: Why do I not need to dereference this??
+    return(clearTurtle(name) && spawnTurtle(name, RESET_X, RESET_Y, 0));
+}
+
+/***TUTORIAL FOR REFERENCE: Actionlib implement an actionserver 7.1******/
+typedef actionlib::SimpleActionServer<software_training_assignment::SeekTurtleAction> Server;
+void move_turtle_callBack(const software_training_assignment::SeekTurtleGoalConstPtr &goal, Server *actionServer)
+{
+    software_training_assignment::SeekTurtleFeedback feedback;
+    software_training_assignment::SeekTurtleResult result;
+
+    geometry_msgs::Twist msg;
+    
+    const float timeInterval = 1; //arbitary time interval?
+    ros::Rate rate(1/timeInterval);
+    const float totalTime = 10;
+    //Declared volatile since I'm not sure how exactly goal is updated...?
+    float volatile deltaX = movingTurtleWP.x - goal->x;
```

Comment:
doesn't need to be volatile. deltaX cannot change outside of this function and is only updated here so no need for it